### PR TITLE
[glsl-in] parse all math functions

### PR DESCRIPTION
**Pull request by Frizi**
Added support for all missing math functions from glsl. Also fixes `mod` to not incorrectly parse as `modf`, which is a separate function.

### CODE REVIEW

File: src/front/glsl/functions.rs [hunk 1]
```diff
@@ -236,38 +251,42 @@ impl Program<'_> {
                             body,
                         )))
                     }
+                    "atan" => match args.len() {
+                        1 => Ok(Some(ctx.add_expression(
```
Comment (kvark) [inline src/front/glsl/functions.rs:255]: nit: the match could return `Expression`, which we'd then add outside of the match

File: tests/in/glsl/math-functions.vert [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#version 450
```
Comment (kvark) [inline tests/in/glsl/math-functions.vert:1]: any reason this doesn't have a corresponding WGSL output snapshot?

Comment (Frizi) [reply tests/in/glsl/math-functions.vert:1]: generation of wgsl output is right now disabled in snapshots, though i'd like to enable it. I can do that in this PR if that's ok.

Comment (kvark) [reply tests/in/glsl/math-functions.vert:1]: fine by me :)

**Comment by kvark** [comment]
> thread 'convert_glsl_folder' panicked at 'called `Result::unwrap()` on an `Err` value: Function { handle: [1], name: "collatz_iterations", error: Expression { handle: [15], error: InvalidBinaryOperandTypes(Modulo, [5], [14]) } }', tests/snapshots.rs:427:10

**Comment by JCapucho** [comment]
can you please rebase on master

**Comment by Frizi** [comment]
Rebased, I see snapshots just got enabled as well, so removed that part.

**Review by JCapucho — APPROVED** [review]
Looks good to me, pinging @kvark since he reviewed originally this PR, do you have anything else you would like fixed or can we merge this?

**Pull request merged**